### PR TITLE
[JENKINS-26940] Print message when installer isn't applicable

### PR DESCRIPTION
**Pull request by daniel-beck**
Example:

```
[Gradle] - Launching build.
Installer "Extract *.zip/*.tar.gz" cannot be used to install "Gradle 2" on the node "Jenkins"
[Gradle] - [ERROR] Can't retrieve the Gradle executable.
```

It seems that users occasionally wonder why their installers "don't work". I'd also print the label expression to be extra helpful, but this may be misleading as `isApplicable(…)` can be overridden.

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Weak 👎 . I had many installers defined for Tool Installations (up to 16 IIRC for label combinations) + multiple tool definitions. Such per-installer logging would pollute my builds much by expected behavior.

My proposal:
* Log info for each installer to system logs on FINE level
* Write only one message to build logs if we're about returning null from the method ("None of installers can be applied")

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
needs change imho

**Comment by oleg-nenashev** [comment]
I'd vote for creating a new low-priority bug/improvement

**Comment by daniel-beck** [comment]
@ndeloof 

> Log info for each installer to system logs on FINE level

Uh, you mean log it in such a way that nobody will ever see it? What's the use of that? Since it's typically something trivial like labels, if you know how to look for the log you know enough to figure out the problem anyway.

---

> Such per-installer logging would pollute my builds much by expected behavior.

How about this -- only print anything when no applicable installer was found, but then, print all of them? Then you'll only have log output in the failure case, i.e. when you need it.

**Comment by oleg-nenashev** [comment]
@daniel-beck 

> How about this -- only print anything when no applicable installer was found, but then, print all of them? Then you'll only have log output in the failure case, i.e. when you need it.

Fine for me. FINE-level logging is not required in such case, of course

**Comment by daniel-beck** [comment]
@oleg-nenashev Please re-review.

File: core/src/main/java/hudson/tools/InstallerTranslator.java [hunk 1]
```diff
@@ -69,8 +73,16 @@ public String getToolHome(Node node, ToolInstallation tool, TaskListener log) th
...
                 }
+            } else {
+                inapplicableInstallersMessages.add(Messages.CannotBeInstalled(
+                        installer.getDescriptor().getDisplayName(),
+                        tool.getName(),
+                        node.getDisplayName()));
             }
         }
+        for (String message : inapplicableInstallersMessages) {
+            log.getLogger().println(message);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tools/InstallerTranslator.java]: Ideally it should be a single message for a correct logging. Especially once we have the External task logging support. Should be fine for now

**Review by oleg-nenashev — APPROVED** [review]

**Review by evieluvsrainbows — APPROVED** [review]

File: core/src/main/java/hudson/tools/InstallerTranslator.java (hunk 1 above)
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tools/InstallerTranslator.java]: We agreed to postpone it for now

**Pull request merged**